### PR TITLE
Created TurretPresets command

### PR DESCRIPTION
**Pull request by ACat701**
Draft pull request for #19

### CODE REVIEW

**Comment by TaylerUva** [comment]
blocked by #20 an #21

**Comment by TaylerUva** [comment]
@ACat701 in your description you will want to say:
close #19
that way it auto closes. the keyword "closes" makes the magic happen.

**Comment by ACat701** [comment]
Closes #19

**Review by TaylerUva — CHANGES_REQUESTED** [review]
Excellent. Now all we need to do is a in a speed limit (don't want the thing to kill itself). We will also need to do `config_kP(0, 1)` on the motors. 

This has to do with a thing called PID (I'll explain in more detail later). Basically the reason it was not working, is for position it does some math P*x+I*y+D*z. If PID are 0 are all zero the equation equals 0. Meaning motor output is 0.

File: src/main/java/frc/robot/subsystems/Hood.java [hunk 1]
```diff
@@ -36,10 +38,18 @@ public double getHoodMotorEncoderCount() {
     return hoodMotor.getSelectedSensorPosition();
   }
 
+  public double getHoodMotorRotation() {
+    return hoodMotor.getSelectedSensorPosition() / 41;
```
Comment (TaylerUva) [inline src/main/java/frc/robot/subsystems/Hood.java:42]: This number is being used in multiple places. What if we want to change it?

File: src/main/java/frc/robot/subsystems/Susan.java [hunk 2]
```diff
@@ -52,4 +52,8 @@ public void setRotationSpeed(double speed) {
         susanMotor.set(ControlMode.PercentOutput, speed);
     }
 
+    public void setRotation(double rotation) {
+        susanMotor.set(ControlMode.Position, rotation * 85);
```
Comment (TaylerUva) [inline src/main/java/frc/robot/subsystems/Susan.java:56]: This number is being used in multiple places. What if we want to change it?

File: src/main/java/frc/robot/subsystems/Hood.java [hunk 3]
```diff
@@ -36,10 +38,18 @@ public double getHoodMotorEncoderCount() {
...
   public void periodic() {
     // This method will be called once per scheduler run
-    SmartDashboard.putNumber("Hood Motor", getHoodMotorEncoderCount());
-
+    SmartDashboard.putNumber("Hood Motor Encoder Count", getHoodMotorEncoderCount());
+    SmartDashboard.putNumber("Hood Rotation", getHoodMotorRotation());
   }
+
+  public void setAngle(double angle) {
+    hoodMotor.set(ControlMode.Position, angle * 41);
```
Comment (TaylerUva) [inline src/main/java/frc/robot/subsystems/Hood.java:53]: This number is being used in multiple places. What if we want to change it?

**Review by TaylerUva — CHANGES_REQUESTED** [review]
Please see my comments

File: src/main/java/frc/robot/subsystems/Susan.java [hunk 4]
```diff
@@ -53,7 +56,8 @@ public void setRotationSpeed(double speed) {
     }
 
     public void setRotation(double rotation) {
-        susanMotor.set(ControlMode.Position, rotation * 85);
+        double r_rotation = rotation;
```
Comment (TaylerUva) [inline src/main/java/frc/robot/subsystems/Susan.java:59]: why create this extra variable?

File: src/main/java/frc/robot/subsystems/Hood.java [hunk 5]
```diff
@@ -50,6 +52,7 @@ public void periodic() {
   }
 
   public void setAngle(double angle) {
-    hoodMotor.set(ControlMode.Position, angle * 41);
+    double r_angle = angle;
```
Comment (TaylerUva) [inline src/main/java/frc/robot/subsystems/Hood.java:55]: why create this extra variable?

File: src/main/java/frc/robot/subsystems/Hood.java [hunk 6]
```diff
@@ -26,6 +26,8 @@ public Hood() {
   // set to factory default
   private void configure() {
     hoodMotor.configFactoryDefault();
+    hoodMotor.configPeakOutputForward(0.4);
+    hoodMotor.configPeakOutputReverse(-0.4);
```
Comment (TaylerUva) [inline src/main/java/frc/robot/subsystems/Hood.java:30]: Have to set k_P here

**Review by TaylerUva — CHANGES_REQUESTED** [review]

**Review by TaylerUva — APPROVED** [review]

**Review by TaylerUva — APPROVED** [review]

**Comment by TaylerUva** [comment]
close #31

**Comment by TaylerUva** [comment]
close #19

**Pull request merged**